Libraries/libs.mk

```diff
@@ -206,9 +206,52 @@ ifeq ($(LIB_EMV), 1)
 EMV_DIR ?= $(LIBS_DIR)/EMV
 
 ifeq ("$(wildcard $(EMV_DIR))","")
-$(error EMV library not found. Please install the EMV package to $(EMV_DIR))
+$(error EMV library not found (Only available via NDA). Please install the EMV package to $(EMV_DIR))
 endif
 
 include $(EMV_DIR)/emv.mk
 endif
 # ************************
+
+# UCL (Disabled by default)
+# Only available via NDA
+# ************************
+LIB_UCL ?= 0
+ifeq ($(LIB_UCL), 1)
+
+TARGET_SEC ?= $(TARGET_UC)
+TARGET_SEC_LC := $(subst M,m,$(subst A,a,$(subst X,x,$(TARGET_SEC))))
+
+PROJ_CFLAGS += -DWORD32 -DM_SDK
+
+ifeq ($(TARGET_SEC), MAX32520)
+PROJ_CFLAGS += -D__arm_m3__ -D__me13
+else ifeq ($(TARGET_SEC), MAX32570)
+PROJ_CFLAGS += -DLARGE_MEMORY -D__me13
+else ifeq ($(TARGET_SEC), MAX32672)
+PROJ_CFLAGS += -DLARGE_MEMORY
+else ifeq ($(TARGET_SEC), MAX32690)
+PROJ_CFLAGS += -DLARGE_MEMORY
+else ifeq ($(TARGET_SEC), MAX32650)
+PROJ_CFLAGS += -DLARGE_MEMORY -D__arm_m4__
+else ifeq ($(TARGET_SEC), MAX32651)
+PROJ_CFLAGS += -DLARGE_MEMORY -D__arm_m4__ -D__max32652
+else ifeq ($(TARGET_SEC), MAX32665)
+PROJ_CFLAGS += -DLARGE_MEMORY
+else ifeq ($(TARGET_SEC), MAX32666)
+PROJ_CFLAGS += -DLARGE_MEMORY
+endif
+
+PROJ_CFLAGS += -D__$(TARGET_SEC_LC)
+
+UCL_DIR ?= $(LIBS_DIR)/UCL
+ifeq ("$(wildcard $(UCL_DIR))","")
+$(error UCL not found (Only available via NDA). Please install the UCL package to $(UCL_DIR))
```

Comment:
@ozersa FYI I added an existence check for `UCL_DIR` here

Reply:
OK